src/app-config.ts

```diff
@@ -1,45 +1,77 @@
+    const sossNodeName = process.env.REACT_APP_SOSS_NODE_NAME || 'romi-dashboard';
 
-if (!process.env.REACT_APP_MOCK && process.env.NODE_ENV !== 'test') {
-  const sossNodeName = process.env.REACT_APP_SOSS_NODE_NAME || 'romi-dashboard';
+    const sossServer = process.env.REACT_APP_SOSS_SERVER;
+    if (!sossServer) {
+      throw new Error('REACT_APP_SOSS_SERVER env variable is needed but not defined');
+    }
 
-  const sossServer = process.env.REACT_APP_SOSS_SERVER;
-  if (!sossServer) {
-    throw new Error('REACT_APP_SOSS_SERVER env variable is needed but not defined');
-  }
+    const trajServer = process.env.REACT_APP_TRAJECTORY_SERVER;
+    if (!trajServer) {
+      throw new Error('REACT_APP_TRAJECTORY_SERVER env variable is needed but not defined');
+    }
 
-  // TODO: get token from some auth service
-  const token = process.env.REACT_APP_SOSS_TOKEN || '';
+    const redirectUri = getRedirectUri();
+    const authConfig: KeycloakConfig = (() => {
+      if (process.env.REACT_APP_AUTH_CONFIG) {
+        return JSON.parse(process.env.REACT_APP_AUTH_CONFIG) as KeycloakConfig;
+      }
+      return {
+        realm: 'master',
+        clientId: 'romi-dashboard',
+        url: 'http://localhost:8080/auth',
+      };
+    })();
+    const authenticator = new DefaultAuthenticator(authConfig, redirectUri.href);
 
-  const trajServer = process.env.REACT_APP_TRAJECTORY_SERVER;
-  if (!trajServer) {
-    throw new Error('REACT_APP_TRAJECTORY_SERVER env variable is needed but not defined');
+    return {
+      authenticator,
+      authRedirectUri: redirectUri.href,
+      appResources: ResourceManager.getResourceConfigurationFile(),
+      transportFactory: () => {
+        const sossToken = process.env.REACT_APP_SOSS_TOKEN || authenticator.sossToken || '';
+        debug.log('authenticating to rmf with token', sossToken);
+        return SossTransport.connect(sossNodeName, sossServer, sossToken);
+      },
+      trajectoryManagerFactory: () => DefaultTrajectoryManager.create(trajServer),
+    };
+  } else {
+    return {
+      authenticator: new FakeAuthenticator(),
+      authRedirectUri: getRedirectUri().href,
+      appResources: ResourceManager.getResourceConfigurationFile(),
+      transportFactory: async () => new FakeTransport(),
+      trajectoryManagerFactory: async () => new FakeTrajectoryManager(),
+    };
   }
-
-  appConfig = {
-    transportFactory: () => SossTransport.connect(sossNodeName, sossServer, token),
-    trajectoryManagerFactory: () => DefaultTrajectoryManager.create(trajServer),
-    appResources: ResourceManager.getResourceConfigurationFile(),
-  };
-} else {
-  appConfig = {
-    transportFactory: async () => new FakeTransport(),
-    trajectoryManagerFactory: async () => new FakeTrajectoryManager(),
-    appResources: ResourceManager.getResourceConfigurationFile(),
-  };
-}
+})();
 
 export default appConfig;
+
+function getRedirectUri(): URL {
+  const redirectUri = new URL(window.location.href);
```

Comment:
I think we should look for a more "React" way for handling the `window.location`. 
What do you think about this? 
https://translate.google.com/translate?sl=en&tl=es&u=https%3A%2F%2Fstackoverflow.com%2Fquestions%2F42253277%2Freact-router-v4-how-to-get-current-route